lib/ts/ink_align.h

```diff
@@ -25,67 +25,78 @@
 #ifndef _ink_align_h_
 #define _ink_align_h_
 
-#include "ts/ink_time.h"
-
-union Alias32 {
-  uint8_t byte[4];
-  int32_t i32;
-  uint32_t u32;
-};
-
-union Alias64 {
-  uint8_t byte[8];
-  int32_t i32[2];
-  uint32_t u32[2];
-  int64_t i64;
-  uint64_t u64;
-  ink_time_t i_time;
-};
+#include <memory>
+
+//
+// workalike to std::align() ... though it's not a great API
+//
+template <typename T_OBJ>
+inline T_OBJ *ats_align(std::size_t alignment, std::size_t size, T_OBJ *&ptr, std::size_t &space)
```

Comment:
Looking at the use-cases, I'm not sure this is a significant improvement over the old INK_ALIGN. It modifies both `ptr` and `space`, but nothing intentionally uses these modified values. Separately, I'm not certain it is necessary for jemalloc. Consider sliding this into its own PR, perhaps re-thinking some of the signatures?